lib/filterable.rb

```diff
@@ -1,4 +1,15 @@
 # frozen_string_literal: true
+require "active_support"
+require "active_support/core_ext/hash/indifferent_access"
+
+require "filterable/base"
+require "filterable/filter"
+require "filterable/filter_set"
+require "filterable/filter/string"
+require "filterable/filter/number"
+require "filterable/filter/date"
+require "filterable/filter/association"
+require "filterable/filter/boolean"
```

Comment:
This reeks tbh, I went 🎯 without loosing time for lots of style and unification of requirements.

A bit of refacto should first clean out a bit how all those classes are defined (lots of duplicated code across all `filter/..` classes that could really use a [template pattern](https://refactoring.guru/design-patterns/template-method) with a `Base` filter, and just override when required.